modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,414 @@
+    group: admission.online.openshift.io
+    groupPriorityMinimum: 1000
+    versionPriority: 15
+    service:
+      name: server
+      namespace: openshift-namespace-reservation
+    version: v1beta1
+
+- apiVersion: admissionregistration.k8s.io/v1beta1  <14>
+  kind: ValidatingWebhookConfiguration
+  metadata:
+    name: namespacereservations.admission.online.openshift.io
+  webhooks:
+  - name: namespacereservations.admission.online.openshift.io
+    clientConfig:
+      service:
+        # reach the webhook via the registered aggregated API
+        namespace: default
+        name: kubernetes
+        path: /apis/admission.online.openshift.io/v1beta1/namespacereservations
+      caBundle: KUBE_CA
+    rules:
+    - operations:
+      - CREATE
+      apiGroups:
+      - project.openshift.io
+      apiVersions:
+      - "*"
+      resources:
+      - projectrequests
+    - operations:
+      - CREATE
+      apiGroups:
+      - ""
+      apiVersions:
+      - "*"
+      resources:
+      - namespaces
+    failurePolicy: Fail
+----
+<1> DaemonSet definition for the webhook server.
+<2> Secret for the service serving cert signer.
+<3> Service account declaration.
+<4> Expose the webhook server within the cluster.
+<5> A custom resource definition for the webhook server.
+<6> `<plural>.<group>` format and reflects custom resource definition `spec` values.
+<7> REST API group name.
+<8> REST API version name.
+<9> Accepted values are `Namespaced` or `Cluster`.
+<10> Plural name to be included in URL.
+<11> Alias seen in `oc` output.
+<12> Camel case reference for resource manifests.
+<13> Configure the admission webhook server also as an aggregated API server.
+<14> Create a validating webhook which which intercepts requests to create namespaces.
+
+. Attribute values to the variables referenced in the above YAML file:
++
+----
+KUBE_CA=$(oc config view --minify=true --flatten -o jsonpath='{range .clusters[*]} {.cluster.certificate-authority-data}')
+----
++
+----
+CERT_DIR=_output/tmp/certs/
+----
+
+. From the same terminal window (i.e. where those variables were defined), deploy the webhook admission server while applying TLS service serving certificate and CA details dynamically:
++
+----
+cat yaml_files/apiserver-list.yaml | \
+    sed "s/TLS_SERVING_CERT/$(base64 ${CERT_DIR}/serving-server.openshift-namespace-reservation.svc.crt | tr -d '\n')/g" | \
+    sed "s/TLS_SERVING_KEY/$(base64 ${CERT_DIR}/serving-server.openshift-namespace-reservation.svc.key | tr -d '\n')/g" | \
+    sed "s/SERVICE_SERVING_CERT_CA/$(base64 ${CERT_DIR}/serving-ca.crt | tr -d '\n')/g" | \
+    sed "s/KUBE_CA/${KUBE_CA}/g" | \
+    oc apply -f -
+----
+
+. Test the admission webhook's functionality:
+.. Reserve a namespace (in this exampled called `reserved-namespace`):
++
+----
+vi yaml_files/reserved_namespace.yaml
+----
++
+----
+apiVersion: online.openshift.io/v1alpha1
+kind: NamespaceReservation
+metadata:
+  name: reserved-namespace
+----
++
+----
```

Comment:
And good job following the guidance to limit each code block to one command, but you should add text/instructions above each one. And have them be substeps.

Reply:
Thank you! I have now amended the procedural steps so that each code block is preceded by a specific instruction.